packages/app/src/cli/models/extensions/specifications/app_proxy.ts

```diff
@@ -0,0 +1,19 @@
+import {createConfigExtensionSpecification} from '../specification.js'
```

Comment:
> Building on top of this schema, is it possible for us to maintain the existing TOML structure?

This is technically possible. For example, we can define cli local `app_preferences` spec schema like this:
```
zod.object({
  app_preferences: zod
    .object({
      url: zod.string().max(255),
    })
    .optional(),
  application_url: zod.string(),
  embedded: zod.boolean(),
})
```
The config content of this extension once this schema is parsed against the toml file will be:
```json
{
  "app_preferences": {
    "url": "https://example.com"
  },
  "application_url": "https://example.com",
  "embedded": true
}
```

> Context: trying to minimize the blast radius of our first milestone and preserve the existing TOML structure if it doesn't create extra tech debt to do so.

First questions is:
- What's the structure of the content expected by the server side?, In case the serve expect a different format, we need to add some kind of support in the CLI to transform the parsed content to the format expected by the server before running the `AppDeploy` request.

The second question is what we need to do in future releases when we need to modify the schema of the app toml to match the proposal made in the [doc](https://docs.google.com/document/d/1EwN1mceHGdvepj8AxrgvchIlenzkiJs3tznfr_O1VSQ/edit):
- Define a second version of each cli local  spec schema to match the new section formats:
- Implement a migration process to transform the current app config toml to the new format
- (optional) support all formats in the same CLI version

Let's summarise the pros and cons of implementing this in two different CLI releases:
- Pros:
  - Migration implementation should not be carried out in the first CLI release
  - Users are not impacted for managing a different app toml file
- Cons:
  - Implement a transform login between the format parsed and the format expected by the server
  - In the second release we need to define a second version of each cli local  spec schema and support previous ones.
  - Managing different schemas versions will make the code brittle

Reply:
@isaacroldan Curious what your thoughts are on this given the CLI as a Client work

Reply:
hey @pt2pham !, regarding the topic of CLI as Client. This PR makes the task to add a new `app_config` spec (a new section in the TOML) to be as simple as adding a new file in the `packages/app/src/cli/models/extensions/specifications` with a content similar to this:
```ts
import {createConfigExtensionSpecification} from '../specification.js'
import {zod} from '@shopify/cli-kit/node/schema'

const PosEmbedmentSchema = zod.object({
  pos: zod
    .object({
      embedded: zod.boolean(),
    })
    .optional(),
})

const spec = createConfigExtensionSpecification({
  identifier: 'pos_embedment',
  schema: PosEmbedmentSchema,
})

export default spec
```
The project to make the CLI as a Client, means that we want to move this definition (mainly the schema definition that will be returned as a JSON schema) as new field in the Core spec definition. In that way, adding a new spec or modifying a current one will require only changes in core an 

For the matter of adding a transform layer between the format in the TOML file and the format the server expect for that app module, we were thinking about adding a transform schema that could look like this for the previous example:

```json
{
    embedded: pos.embedded
}
```
with this transformation schema definition we would have this forma in the TOML file:
```toml
[pos]
embedded = true
```
but the content the CLI will send as the `config` of the extension registration will be:
{
  embedded: true
}

in this example we are flattening the object inside the TOML when the content is sent:
The same strategy could be followed to support the current TOML file schema to send the content of the new `app_config` modules. 
For the case of the new `app_ui` app module
_TOML schema definition:_
```ts
zod.object({
  app_preferences: zod
    .object({
      url: zod.string().max(255),
    })
    .optional(),
  application_url: zod.string(),
  embedded: zod.boolean(),
})
```
_ Transform schema_
```json
{
  app_url: application_url
  embedded: embedded,
  preferences_url: app_preferences.url
}
```
Regarding the CLI as client this transform definition can also be returned from the server side. In this example we should explore if it possible to declare complex transformation rules, ie: `embedded: zod.boolean() => ui: zod.enum('embedded', 'external')`